src/pages/MainPage/components/ProceedContainer/components/SumaryContainer/components/SumaryBody/components/TableBodyContainer/index.js

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+
+import {
+  selectBasket,
+  setSubtotal,
+  setGrandTotal,
+  setShipping,
+} from "features/basket/basketSlice";
+import { Button } from "components";
+import { TableItem } from "./components";
+import * as Styles from "./styles";
+
+const TableBodyContainer = () => {
+  const dispatch = useDispatch();
+  const basket = useSelector(selectBasket);
+
+  useEffect(() => {
+    dispatch(setSubtotal(subtotal()));
+    dispatch(setShipping(shipping));
+    dispatch(setGrandTotal(subtotal() === 0 ? 0 : subtotal() + shipping));
+  }, [basket]);
+
+  const subtotal = () => {
```

Comment:
const getSubtotal = () => basket.items.reduce((acc, item) => acc + item.price * item.value, 0)

ale lepiej zrobić do tego selector, po to żeby tutaj mieć to jako zmienną, a nie funkcję, odpalaną kilka razy w różnych miejsach. Użyj createSelector z reselect i zrób selektor, który to zwraca. To będzie coś w tym stylu:

export const selectBasketSubtotal = createSelector(
  selectBasketItems,
  basketItems => basketItems.reduce((acc, item) => acc + item.price * item.value, 0)
)

i tutaj już const subtotal = useSelector(selectBasketSubtotal)

Reply:
Done.

Reply:
Zrobione ale mam pytanie, reselect w teori może zastąpić dispatch bo po co dispatchować jeśli zawsze może obliczyć to za nas reselect. Ale z drugiej strony rozumiem że oszczędzamy pamięć jeśli dispatchujemy bo reselekt wywołuje się za każdym razem a dispatch raz ? Nie wiem czy dobrze to wytłumaczyłem ale zastanawiam się nad zastąpieniem dispatcha reselectem w takich wypatkach tylko pytanie  czy nie niesie to za sobą jakiś zagrożeń ? 
Teraz użyłem reselecta jako wartości do dispatcha a odbieram innym selectorem "basket".

Reply:
During...